hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/bucket/ListBucketHandler.java

```diff
@@ -50,13 +53,33 @@ protected void execute(OzoneClient client, OzoneAddress address)
     OzoneVolume vol = client.getObjectStore().getVolume(volumeName);
     Iterator<? extends OzoneBucket> bucketIterator =
         vol.listBuckets(listOptions.getPrefix(), listOptions.getStartItem());
-
-    int counter = printAsJsonArray(bucketIterator, listOptions.getLimit());
+    int counter = printBuckets(bucketIterator, listOptions.getLimit());
 
     if (isVerbose()) {
       out().printf("Found : %d buckets for volume : %s ", counter, volumeName);
     }
   }
 
+  private int printBuckets(Iterator<? extends OzoneBucket> bucketIterator,
+                           int limit) {
+    int counter = 0;
+    final ArrayNode arrayNode = JsonUtils.createArrayNode();
```

Comment:
We don't need to manually create JSON-specific objects.  Can create a `List<Object>` instead, and depending on whether each item is a bucket or a link, can add `bucket` itself or `new LinkBucket(bucket)`.  Then the list can be printed via `printObjectAsJson(list)`.